qcodes/instrument_drivers/Lakeshore/Model_325.py

```diff
@@ -0,0 +1,251 @@
+        >>> ...[64, 32]  # This is correct because 96=64+32
+        """
+        if number in components:
+            terms = [number]
+        else:
+            terms = []
+            comp = np.sort(components)[::-1]
+            comp = comp[comp <= number]
+
+            while len(comp):
+                c = comp[0]
+                number -= c
+                terms.append(c)
+
+                comp = comp[comp <= number]
+
+        return terms
+
+
+class Output_325(InstrumentChannel):
+    def __init__(self, parent, name, channel):
+        super().__init__(parent, name)
+        self._channel = channel
+
+        self.add_parameter(
+            "control_mode",
+            get_cmd=f"CMODE? {self._channel}",
+            set_cmd=f"CMODE {self._channel},{{}}",
+            val_mapping={
+                "Manual PID": "1",
+                "Zone": "2",
+                "Open Loop": "3",
+                "AutoTune PID": "4",
+                "AutoTune PI": "5",
+                "AutoTune P": "6"
+            }
+        )
+
+        self.add_parameter(
+            "input_channel",
+            vals=Enum("A", "B", "a", "b"),
+            parameter_class=GroupParameter
+        )
+
+        self.add_parameter(
+            "unit",
+            vals=Enum("Kelvin", "Celsius", "Sensor Units"),
+            val_mapping={
+                "Kelvin": "1",
+                "Celcius": "2",
+                "Sensor Units": "3"
+            },
+            parameter_class=GroupParameter
+        )
+
+        self.add_parameter(
+            'powerup_enable',
+            val_mapping={True: 1, False: 0},
+            parameter_class=GroupParameter
+        )
+
+        self.add_parameter(
+            "output_metric",
+            vals=Enum("current", "power"),
+            val_mapping={
+                "current": "1",
+                "power": "2",
+            },
+            parameter_class=GroupParameter
+        )
+
+        Group(
+            [self.input_channel, self.unit, self.powerup_enable,
+             self.output_metric],
+            set_cmd=f"CSET {self._channel}, {{input_channel}}, {{unit}}, "
+                    f"{{powerup_enable}}, {{output_metric}}",
+            get_cmd=f"CSET? {self._channel}"
+        )
+
+        self.add_parameter(
+            'P',
+            vals=Numbers(0, 1000),
+            get_parser=float,
+            parameter_class=GroupParameter
+        )
+
+        self.add_parameter(
+            'I',
+            vals=Numbers(0, 1000),
+            get_parser=float,
+            parameter_class=GroupParameter
+        )
+
+        self.add_parameter(
+            'D',
+            vals=Numbers(0, 1000),
+            get_parser=float,
+            parameter_class=GroupParameter
+        )
+
+        Group(
+            [self.P, self.I, self.D],
+            set_cmd=f'PID {self._channel}, {{P}}, {{I}}, {{D}}',
+            get_cmd=f'PID? {self._channel}'
+        )
+
+        if self._channel == 0:
```

Comment:
what is this `0` channel/heater?

I also don't see it being created in the driver class. Shall it be removed, if it's not used, or shall it be properly added?

Reply:
![image](https://user-images.githubusercontent.com/10454668/45479910-04bfb100-b748-11e8-8a07-4d3dd366fe6c.png)

Reply:
The mistake here is that a string (self._channel) is being compared to a number (0)